presto-tests/src/test/java/io/prestosql/memory/TestMemoryManager.java

```diff
@@ -327,6 +329,50 @@ public void testClusterPools()
         }
     }
 
+    @Test(timeOut = 240_000)
+    public void testSessionQueryMemoryPerNodeLimit()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = createQueryRunner(TINY_SESSION, ImmutableMap.of())) {
```

Comment:
You should run two queries. One that passes without memory limit and one that fails once you apply memory limit session property. Take a look at `testResourceOverCommit` test.

Reply:
@sopel39 How about extend `AbstractTestQueryFramework` for these tests ?

Reply:
@sopel39 I do run the query twice at line 337 and line 345. Or I might be misunderstanding the ask.

Reply:
discussed offline, this is already written in the suggested format